src/App.tsx

```diff
@@ -1,24 +1,294 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState, useCallback, useEffect, useMemo, useRef,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import {
+  getTodos, postTodo, deleteTodo, patchTodo, USER_ID,
+} from './api/todos';
+import TodoList from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import Footer from './components/Footer';
+import HeaderForm from './components/HeaderForm';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [query, setQuery] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [loading, setLoading] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number[]>([0]);
+  const [isErrorVisible, setIsErrorVisible] = useState<boolean>(Boolean);
```

Comment:
you can use only one error state. `isErrorVisible` can be calculated based on `errorText`
```suggestion
  const isErrorVisible = errorText !== '';
```